pkg/interface/src/views/landscape/components/ResourceSkeleton.tsx

```diff
@@ -121,26 +124,69 @@ export function ResourceSkeleton(props: ResourceSkeletonProps): ReactElement {
     </TruncatedText>
   );
 
-  const WriterControls = () => (
-    <Link to={resourcePath('/new')}>
-      <Text bold pr='3' color='blue'>
-        + New Post
-      </Text>
-    </Link>
-  );
+  const ExtraControls = () => {
+    if (workspace === '/messages' && group.hidden)
+      return (
+        <Dropdown
+          flexShrink={0}
+          dropWidth='300px'
+          width='auto'
+          alignY='top'
+          alignX='right'
+          options={
+            <Col
+              backgroundColor='white'
+              border={1}
+              borderRadius={2}
+              borderColor='lightGray'
+              color='washedGray'
+              boxShadow='0px 0px 0px 3px'
+            >
+              {group.members.size >= 3 ? (
+                <MessageInvite association={association} api={api} />
+              ) : null}
+              {group.members.size === 2 ? (
+                <NewChannel
+                  api={props.api}
+                  history={history}
+                  workspace={{ type: 'messages' }}
+                  borderRadius={2}
+                  existingMembers={without(
+                    Array.from(group.members),
+                    window.ship
+                  )}
+                />
+              ) : null}
+            </Col>
+          }
+        >
+          <Text bold pr='3' color='blue'>
+            + Add Ship
+          </Text>
+        </Dropdown>
+      );
+    if (canWrite)
+      return (
+        <Link to={resourcePath('/new')}>
+          <Text bold pr='3' color='blue'>
+            + New Post
+          </Text>
+        </Link>
+      );
+    return <></>;
+  };
 
   const MenuControl = () => (
     <Link to={`${baseUrl}/settings`}>
       <Icon icon='Menu' color='gray' pr='2' />
     </Link>
   );
 
-  const actRef = actionsRef.current;
-  let actionsWidth = 0;
-
-  if (actRef) {
-    actionsWidth = actRef.clientWidth;
-  }
+  useEffect(() => {
+    if (actionsRef.current) {
+      setActionsWidth(actionsRef.current.clientWidth);
+    }
+  }, [actionsRef]);
```

Comment:
Wouldn't it be easier to just use a callback ref to call the setter directly?

Reply:
Far easier, thank you